services/cargohold/test/integration/API/Util.hs

```diff
@@ -18,30 +18,59 @@
 module API.Util where
 
 import Bilge hiding (body, host, port)
+import qualified Bilge
 import CargoHold.Options
 import CargoHold.Run
 import qualified Codec.MIME.Parse as MIME
 import qualified Codec.MIME.Type as MIME
-import Control.Lens
+import Control.Lens hiding ((.=))
 import Control.Monad.Codensity
+import Data.Aeson (object, (.=))
 import Data.ByteString.Builder
+import qualified Data.ByteString.Char8 as C
 import Data.ByteString.Conversion
 import qualified Data.ByteString.Lazy as Lazy
 import Data.Domain
 import Data.Id
 import Data.Qualified
-import Data.Text.Encoding (decodeLatin1)
+import Data.Text.Encoding (decodeLatin1, encodeUtf8)
 import qualified Data.UUID as UUID
+import Data.UUID.V4 (nextRandom)
 import Federator.MockServer
 import Imports hiding (head)
 import qualified Network.HTTP.Media as HTTP
 import Network.HTTP.Types.Header
 import Network.HTTP.Types.Method
 import Network.Wai.Utilities.MockServer
+import Safe (readNote)
 import TestSetup
 import Util.Options
 import Wire.API.Asset
 
+-- Copied wholesale from gundeck/test/integration/API.hs
+-- This is needed because it sets up the email on the user, verifiying it.
+-- The changes to the asset routes forbidding non-verified users from uploading
+-- assets breaks a lot of existing tests.
+randomUser :: TestM UserId
+randomUser = do
+  (Endpoint (encodeUtf8 -> eHost) ePort) <- view tsBrig
+  e <- liftIO $ mkEmail "success" "simulator.amazonses.com"
+  let p =
+        object
+          [ "name" .= e,
+            "email" .= e,
+            "password" .= ("secret-8-chars-long-at-least" :: Text)
+          ]
+  r <- post (Bilge.host eHost . Bilge.port ePort . path "/i/users" . json p)
+  pure
+    . readNote "unable to parse Location header"
+    . C.unpack
+    $ getHeader' "Location" r
+  where
+    mkEmail loc dom = do
+      uid <- nextRandom
+      pure $ loc <> "+" <> UUID.toText uid <> "@" <> dom
+
```

Comment:
We created the new test framework so we can avoid this kind of wholesale copying.

Reply:
This is needed to support existing legacy tests. The change to the API breaks them, as they were using random IDs that weren't marked as verified. This function sets up a new user to be verified.

Reply:
I think that would be a good idea, but would likely warrant its own ticket.

Reply:
Can we paste the ticket here for posterity?

Reply:
I created the ticket: https://wearezeta.atlassian.net/browse/WPB-5382